numpyro/hmc_util.py

```diff
@@ -239,7 +239,7 @@ def init_fn(z, rng, step_size=1.0, inverse_mass_matrix=None, mass_matrix_size=No
             else:
                 inverse_mass_matrix = np.identity(mass_matrix_size)
 
-        if find_reasonable_step_size is not None:
+        if adapt_step_size and (find_reasonable_step_size is not None):
```

Comment:
Can we not just fix the `find_reasonable_step_size` function to the default implementation and not have to do checks subsequently, i.e. add the default to the kwarg in the constructor `find_reasonable_step_size=_find..`?

Reply:
I think it would be rare for users to provide a custom implementation for this, so we might as well just use the default which they can override if they wish.